org.scala-ide.sdt.core/src/org/scalaide/core/compiler/DeclarationPrinter.scala

```diff
@@ -67,12 +67,10 @@ abstract class DeclarationPrinter extends HasLogger {
     /* Inspired from Symbol and HasFlags, but take into account the mask access modifiers.
      * The original would always display access modifiers, even when the mask is `0L`.
      */
-    def flagString(sym: Symbol, basis: Long): String = {
-      import sym.{ accessString => _, _ }
+    def flagString(sym: Symbol): String = {
       import Flags._
-      val access = accessString(sym)
-
-      val nonAccess = flagBitsToString(basis & ~AccessFlags)
+      val access = if (showAccessor) accessString(sym) else ""
+      val nonAccess = sym.flagBitsToString(sym.flags & flagMask & ~AccessFlags)
 
```

Comment:
generally your refactoring is pretty ok, but I would implement `flagString` as
```
    def flagString(sym: Symbol): String = {
      import Flags._
      val access = if ((flagMask & AccessFlags | flagMask) != 0) accessString(sym) else ""
      val nonAccess = sym.flagBitsToString(sym.flags & flagMask & ~AccessFlags)

      if (access == "") nonAccess
      else if (nonAccess == "") access
      else nonAccess + " " + access
    }
```

and then you don't need `showAccessor` flag. Is it ok (this my change passes tests but I don't know if there are other scenarios)?